src/main/scala/beam/router/Modes.scala

```diff
@@ -261,6 +262,12 @@ object TourModes {
 
     import BeamTourMode._
 
+    def vehicleSharedOrNot(vehicles: Vector[VehicleOrToken]): Boolean = {
```

Comment:
There's some scala magic that could make this neater -- something like `vehicles.any(_.vehicle.isSharedVehicle)` I think. But either way, I think we should either use this function to filter the allowed beam modes here or just move this function over to ChoosesMode where it's used, probably no reason for it to be in the `Modes` class otherwise

Reply:
Interesting! Thank you for letting me know!

Reply:
actually I think it's `forall` rather than `any`

Reply:
Here's a suggested way of doing all the filtering within `BeamTourMode`

    def allowedBeamModesGivenAvailableVehicles(
      vehicles: Vector[VehicleOrToken],
      firstOrLastLeg: Boolean
    ): Seq[BeamMode] = {
      val relevantModes = if (firstOrLastLeg) { allowedBeamModesForFirstAndLastLeg }
      else allowedBeamModes
      relevantModes.filter { mode =>
        vehicles.exists { vehOrToken =>
          !vehOrToken.vehicle.isSharedVehicle && (vehOrToken.streetVehicle.mode == mode)
        }
      }
    }

Reply:
addressed!